pkg/arvo/app/launch.hoon

```diff
@@ -45,19 +53,36 @@
     def   ~(. (default-agent this %|) bol)
 ++  on-init
   ^-  (quip card _this)
-  :_  this
+  :_  this(state *state-one)
   [%pass / %arvo %e %connect [~ /] %launch]~
 ::
 ++  on-save  !>(state)
 ::
 ++  on-load
   |=  old=vase
-  `this(state !<(state-zero old))
+  ^-  (quip card _this)
+  =/  old-state  !<(versioned-state old)
+  :-  ~
+  ?-    -.old-state
+      %0
+    %=  this
+      state  [%1 tiles.old-state data.old-state path-to-tile.old-state %.n]
+    ==
+  ::
+      %1  this(state old-state)
+  ==
 ::
 ++  on-poke
   |=  [mar=mark vas=vase]
   ^-  (quip card _this)
   ?+    mar  (on-poke:def mar vas)
+      %json
+    ?>  (team:title our.bol src.bol)
+    =/  jon  !<(json vas)
+    :-  ~
+    ?.  =(jon [%s 'disable welcome message'])
+      this
+    this(first-time %.n)
```

Comment:
This should, uh, probably be a `%launch-action`? But that isn't currently tagged with anything, which is... unfortunate, because that means adding this to the action type means changing its use everywhere.

If we're gonna be fine with just punching bespoke json into this, can we at least make it something approximating what the action json would be? Perhaps `[%o ['dismiss-welcome'^~ ~ ~]]` aka `{'dismiss-welcome': null}` aka `[%dismiss-welcome ~]`?